scripts/initialize

```diff
@@ -1,77 +1,145 @@
+    echo "You are not a sudoer!"
+fi
+# Check for group admin
+if groups | grep "\<admin\>" &> /dev/null; then
+    echo "You are an admin!"
+    export SUDO="sudo -s"
+else
+    echo "You are not an admin!"
+fi
+#DETECT ENV VARS
 OS=$(uname)
 OS_VERSION=$(uname -r)
 UNAME_M=$(uname -m)
+UNAME_P=$(uname -p)
 ARCH=$(uname -m)
+if [[ "$OS" != "Linux"* ]]; then
+    if hash brew; then
+        BREW=$(which brew)
+        BREW_AWK=$(brew --prefix awk)
+        BREW_GAWK=$(brew --prefix gawk)
+        export HOMEBREW_NO_ENV_HINTS=0
+        export HOMEBREW_NO_INSTALL_CLEANUP=false
+    fi
+fi
+PYTHON3=$(which python3)
+PIP=$(which pip)
+PIP3=$(which pip3)
+if [ -z "$SUDO" ]; then
+    DASH_U="-U"
+else
+    DASH_U=
+fi
 AWK=$(which awk)
 GAWK=$(which gawk)
-export OS
-export OS_VERSION
-export UNAME_M
-export ARCH
-export AWK
-export GAWK
+
+echo "OS=$OS"
+echo "OS_VERSION=$OS_VERSION"
+echo "UNAME_M=$UNAME_M"
+echo "UNAME_P=$UNAME_P"
+echo "ARCH=$ARCH"
+echo "BREW=$BREW"
+echo "PYTHON3=$PYTHON3"
+echo "PIP=$PIP"
+echo "PIP3=$PIP3"
+echo "DASH_U=$DASH_U"
+echo "AWK=$AWK"
+echo "GAWK=$GAWK"
+echo "BREW_AWK=$BREW_AWK"
+echo "BREW_GAWK=$BREW_GAWK"
+
+if [ "${UNAME_M}" = "x86_64" ]; then
+    if [ "${OS}" = "Darwin"* ]; then
+        if [ "$(sysctl -in sysctl.proc_translated)" = "1" ]; then
+            echo "Running on Rosetta 2"
+            ROSETTA=1
+            export ROSETTA
+        else
+            echo "Running on native Intel"
+        fi
+    fi
+elif [ "${UNAME_M}" = "arm64" ]; then
+    echo "Running on arm64"
+elif [ "${UNAME_M}" = "aarch64" ]; then
+    echo "Running on aarch64"
+else
+    echo "Unknown architecture: ${UNAME_M}"
+fi
 
 if [[ ! -z "$GAWK" ]]; then
     AWK=$(which gawk)
     export AWK
-    echo $AWK
 fi
 
-export PATH=$PATH:/usr/local/opt/python@${python_version_major}.${python_version_minor}/Frameworks/Python.framework/Versions/${python_version_major}.${python_version_minor}/bin
-
-report() {
-echo OS:
-echo "$OS" | "$AWK" '{print tolower($0)}'
-echo OS_VERSION:
-echo "$OS_VERSION" | "$AWK" '{print tolower($0)}'
-echo UNAME_M:
-echo "$UNAME_M" | "$AWK" '{print tolower($0)}'
-echo ARCH:
-echo "$ARCH" | "$AWK" '{print tolower($0)}'
-echo OSTYPE:
-echo "$OSTYPE" | "$AWK" '{print tolower($0)}'
+function initialize-add-python3-to-path(){
+#ADD python3 to PATH
+if [[ ! -z "${python_version_major}" ]]; then
+    if [[ ! -z "${python_version_minor}" ]]; then
+        export PATH=$PATH:/usr/local/opt/python@${python_version_major}.${python_version_minor}/Frameworks/Python.framework/Versions/${python_version_major}.${python_version_minor}/bin
+    fi
+fi
+}
+
+function initialize-use-apk(){
+# alpine linux... not android
+    echo "more apk support needed"
+    apk add "$AWK"
+    apk add python3 py3-pip
+    python3 -m pip install -r requirements.txt || \
+    python3 -m pip install -r ../requirements.txt
+
 }
+function initialize-use-apt(){
 
-checkbrew() {
+DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt-get -y install tzdata
```

Comment:
I found this useful in other environments...
some packages will prompt for timezone when installing if this isn't installed...